src/hotspot/share/opto/mulnode.cpp

```diff
@@ -610,6 +610,14 @@ Node *AndINode::Ideal(PhaseGVN *phase, bool can_reshape) {
     return progress;
   }
 
+  // Convert "(~a) & (~b)" into "~(a | b)"
+  if (in(1)->Opcode() == Op_XorI
+      && phase->type(in(1)->in(2)) == TypeInt::MINUS_1
+      && in(2)->Opcode() == Op_XorI
+      && in(1)->in(2) == in(2)->in(2)) {
```

Comment:
minor code style issue: please take the `&&` to the end of the line. That is what I usually see. It also makes reading the lines easier, as they are aligned with the first line.

Reply:
```suggestion
      && phase->type(in(2)->in(2)) == TypeInt::MINUS_1) {
```
Could be nice for symmetry.